The system printf fails the /string/test-string-nul-handling test on
11.3 as of glib 2.76.4.  I cannot reproduce this in isolation.

Switching to the glib-supplied printf resolves the issue.

--- glib-2.76.4/meson.build.~4~	2023-08-10 12:22:11.736866190 +0000
+++ glib-2.76.4/meson.build	2023-08-10 12:50:01.703140583 +0000
@@ -1219,7 +1219,7 @@
 endif
 
 # Our printf is 'good' only if vsnpintf()/snprintf()/printf() supports C99 well enough
-use_system_printf = have_good_vsnprintf and have_good_snprintf and have_good_printf
+use_system_printf = false
 glib_conf.set('USE_SYSTEM_PRINTF', use_system_printf)
 glibconfig_conf.set('GLIB_USING_SYSTEM_PRINTF', use_system_printf)
 
